packages/contracts/contracts/optimistic-ethereum/OVM/chain/OVM_CanonicalTransactionChain.sol

```diff
@@ -971,75 +973,97 @@ contract OVM_CanonicalTransactionChain is iOVM_CanonicalTransactionChain, Lib_Ad
+     * @param _laterContext The batch context whose time values should come AFTER all of _earlierContext's subsequent queue transactions
+     * @param _firstfirstSubsequentQueueIndex Index of the queue element we are validating came later than the _earlierContext
      * @param _queueRef The storage container for the queue.
      */
-    function _validateContextBeforeEnqueue(
-        BatchContext memory _context,
-        uint40 _queueIndex,
+    function _validateEnqueuesBetweenContexts(
+        BatchContext memory _earlierContext,
+        BatchContext memory _laterContext,
+        uint40 _firstfirstSubsequentQueueIndex,
         iOVM_ChainStorageContainer _queueRef
     )
         internal
         view
     {
-            Lib_OVMCodec.QueueElement memory nextQueueElement = _getQueueElement(
-                _queueIndex,
+            Lib_OVMCodec.QueueElement memory earliestQueueElementBetween = _getQueueElement(
+                _firstfirstSubsequentQueueIndex,
                 _queueRef
             );
 
+            // Assume the earliest and latest are the same, update the latest if not
+            Lib_OVMCodec.QueueElement memory latestQueueElementBetween = earliestQueueElementBetween;
+            if(_earlierContext.numSubsequentQueueTransactions > 1){
+                latestQueueElementBetween = _getQueueElement(
+                    _firstfirstSubsequentQueueIndex + uint40(_earlierContext.numSubsequentQueueTransactions) - 1,
+                    _queueRef
+                );
+            }
+
             // If the force inclusion period has passed for an enqueued transaction, it MUST be the next chain element.
             require(
-                block.timestamp < nextQueueElement.timestamp + forceInclusionPeriodSeconds,
+                block.timestamp < earliestQueueElementBetween.timestamp + forceInclusionPeriodSeconds,
                 "Previously enqueued batches have expired and must be appended before a new sequencer batch."
             );
 
             // Just like sequencer transaction times must be increasing relative to each other,
             // We also require that they be increasing relative to any interspersed queue elements.
             require(
-                _context.timestamp <= nextQueueElement.timestamp,
+                _earlierContext.timestamp <= earliestQueueElementBetween.timestamp,
                 "Sequencer transaction timestamp exceeds that of next queue element."
             );
 
             require(
-                _context.blockNumber <= nextQueueElement.blockNumber,
+                _earlierContext.blockNumber <= earliestQueueElementBetween.blockNumber,
                 "Sequencer transaction blockNumber exceeds that of next queue element."
             );
+
+            require(
+                _laterContext.timestamp >= latestQueueElementBetween.timestamp,
+                "Sequencer transaction timestamp must be greater than last queue element appended."
```

Comment:
Is it worth updating the strings here to be more consistent with the above?  `Sequencer transaction blockNumber is less than that of previous queue element.`